jenkins/Jenkinsfile

```diff
@@ -0,0 +1,477 @@
+    // so disable for now.
+
+    // def job = Jenkins.instance.getItemByFullName (env.JOB_NAME)
+    // for (build in job.builds) {
+    //     if (!build.isBuilding ())
+    //         continue
+
+    //     if (build.number > currentBuild.number) {
+    //         error ("There is already a newer build in progress (#${build.number})")
+    //     } else if (build.number < currentBuild.number) {
+    //         def exec = build.getExecutor ()
+    //         if (exec == null) {
+    //             echo ("No executor for build ${build.number}")
+    //         } else {
+    //             exec.interrupt (Result.ABORTED, new CauseOfInterruption.UserInterruption ("Aborted by build #${currentBuild.number}"))
+    //             echo ("Aborted previous build: #${build.number}")
+    //         }
+    //     }
+    // }
+}
+
+github_pull_request_info = null
+def githubGetPullRequestInfo ()
+{
+    if (github_pull_request_info == null && isPr) {
+        withCredentials ([string (credentialsId: 'macios_github_comment_token', variable: 'GITHUB_PAT_TOKEN')]) {
+            def url = "https://api.github.com/repos/${repository}/pulls/${env.CHANGE_ID}"
+            def outputFile = ".github-pull-request-info.json"
+            try {
+                sh ("curl -vf -H 'Authorization: token ${GITHUB_PAT_TOKEN}' --output '${outputFile}' '${url}'")
+                github_pull_request_info = readJSON (file: outputFile)
+                echo ("Got pull request info: ${github_pull_request_info}")
+            } finally {
+                sh ("rm -f ${outputFile}")
+            }
+        }
+
+    }
+    return github_pull_request_info
+}
+
+github_pull_request_labels = null
+def githubGetPullRequestLabels ()
+{
+    if (github_pull_request_labels == null) {
+        github_pull_request_labels = []
+        if (isPr) {
+            def pinfo = githubGetPullRequestInfo ()
+            def labels = pinfo ["labels"]
+            if (labels != null) {
+                for (int i = 0; i < labels.size (); i++) {
+                    def label = labels [i]
+                    github_pull_request_labels.add (label ["name"])
+                }
+            }
+            echo ("Found labels ${github_pull_request_labels} for the pull request.")
+        }
+    }
+    return github_pull_request_labels
+}
+
+def githubAddComment (url, markdown)
+{
+    def json = groovy.json.JsonOutput.toJson ([body: markdown])
+    def jsonFile = "${workspace}/xamarin-macios/jenkins/commit-comments.json"
+    try {
+        writeFile (file: "${jsonFile}", text: "${json}")
+        sh ("cat '${jsonFile}'")
+        withCredentials ([string (credentialsId: 'macios_github_comment_token', variable: 'GITHUB_COMMENT_TOKEN')]) {
+            sh ("curl -i -H 'Authorization: token ${GITHUB_COMMENT_TOKEN}' ${url} --data '@${jsonFile}'")
+        }
+    } finally {
```

Comment:
Same as previous, do we want to log exceptions?

Reply:
Same answer :smile:

Reply:
Fair enough ;)